### DISCUSSION TITLE
AG plugin crash with Studio One

### DISCUSSION THREAD

**Discussion by Henri75** [apohl79/audiogridder#994] [General]
It started with beta 10 (never happened before, but it may be related to new plugins I use).

On Mac, some AG plugin inserts are crashing (you  got the red smiley I never saw before in Studio One plugin window).

Crash is random, you load a session it works. Keep the AG insert or play with it, switch between inserts, or just let it opened, and it will crash randomly sooner or later)
Reload the session, and it works again.

One chain I got issues with is : Gulfoss master + P42 climax.

All on MAC M1 native (Studio One, client, server), latest versions of everything

**Comment by apohl79** [comment]
Do you have crash reporting enabled on the server side? I will hopefully see some crashes in that case and might get some conclusions..

**Comment by Henri75** [comment]
I will enable logs and send them to you.



Le dim. 21 août 2022 à 19:53, Andreas Pohl ***@***.***> a
écrit :

**Comment by Henri75** [comment]
Hi Andreas,

So today I tried to change the MTU on all my devices to Jumbo 9000 (trying to debug some drops/crackles related to cpu spikes with AG and S1 running 2 mac servers with 10% cpu load and 15 MB/s).
All my devices works fine at MTU 9000 (over 950 mb/s in full duplex and low cpu) except AG.
Interestingly, all AG plugins started to crash randomly in S1. Not sure if this is the same issue I encountered the  last time, but it may.

I archived the logs of the test session and upload it on Github 

There is no obvious reasons for the several crashes of plugins (some where opened, others not).

I noticed too three interesting facts (that may be related to this one and some cpu spikes that are happenning with every AG version on large sessions ) 

- with a MTU of 9000 the remote plugins displayed are flickering. There is too some display/network issues that is probably related to this. At the same time I had 2 opened sessions on the same remote servers with anydesk that were perfectly smooth.

- with a MTU of 9000 I got a lot more drops in my S1 sessions compared to regulat 1500 MTU (whereas the cpu load is decreased on both the mac client < 50% and mac servers <10%, the latency is reduced and the throughput increased). So it seems the drops is related to some issues in AG, which get worst with higher network throughput and lower latency.
I had regular crackles/drops on large sessions for a while. When I disable all AG plugins the sessions are fine even when there is high cpu usage. So it seems there is something related to AG, despite the average cpu load on M1 remote servers (<50%) and average global bandwith (300-400 mb/sec)

[AudioGridder.zip](https://github.com/apohl79/audiogridder/files/9482979/AudioGridder.zip)

**Reply by apohl79** [reply]
In the plugin log I see the plugins were loading and unloading properly. So what do you mean by plugin crash?

**Comment by Henri75** [comment]
Here is an example of the error message when Studio one disable globally AG because of too many crashes.

it never happened in previous versions.
![Screen Shot 2022-09-07 at 20 30 16](https://user-images.githubusercontent.com/56123216/188952314-28ab63a4-4107-4b91-ab1d-33cfd73b6f00.png)

**Comment by Henri75** [comment]
During that last crash of the AG plugin, I checked and the plugin was still loaded and the GUI  open on the remote mac server

**Comment by Henri75** [comment]
Here is a screenshot when I load a session that has this kind of problem 
![Screen Shot 2022-09-07 at 20 40 07](https://user-images.githubusercontent.com/56123216/188953934-be531bc4-281e-44a1-8497-b64f60549aa9.png)
: 
You can see the red smiley on the AG plugin window In studio (which mean crash/disabled plugin)
The plugin is open and working on the remote server.

in the tray monitor everything seems to be ok !
![Screen Shot 2022-09-07 at 20 42 27](https://user-images.githubusercontent.com/56123216/188954056-6b94ea11-ce14-48cd-a18a-f613cfcbcb8e.png)

It seems to happen with a few inserts/plugins (from different editors). not all of them, and not systematically. 

But when one AG plugin instance crashes your projects is not usable anymore :(

**Reply by apohl79** [reply]
Have you tried to increase the buffer size? I don't know under what conditions S1 would stop a plugin. But the only impact jumbo frames should have is latency. It might affect it negatively.
